idea_plugin/build.gradle.kts

```diff
@@ -15,40 +16,42 @@
  */
 
 // https://github.com/JetBrains/intellij-platform-gradle-plugin/releases
-plugins { id("org.jetbrains.intellij") version "1.17.3" }
-
-apply(plugin = "org.jetbrains.intellij")
+plugins {
+  id("org.jetbrains.intellij.platform") version "2.0.1"
+}
 
-apply(plugin = "java")
+repositories {
+  mavenCentral()
 
-repositories { mavenCentral() }
+  intellijPlatform {
+    defaultRepositories()
+  }
+}
 
 // https://github.com/google/google-java-format/releases
-val googleJavaFormatVersion = "1.22.0"
+val googleJavaFormatVersion = "1.23.0"
```

Comment:
let's save this upgrade for a different commit... there are a few other things that need to change with a version upgrade (like the release notes), so let's just keep this PR for the build configuration changes.